glide-core/src/client/value_conversion.rs

```diff
@@ -287,6 +288,28 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        // Used by BZPOPMIN/BZPOPMAX, which return an array consisting of the key of the sorted set that was popped, the popped member, and its score.
+        // RESP2 returns the score as a string, but RESP3 returns the score as a double. Here we convert string scores into type double.
+        ExpectedReturnType::KeyWithMemberAndScore => match value {
+            Value::Nil => Ok(value),
+            Value::Array(ref array) if array.len() == 3 && matches!(array[2], Value::Double(_)) => {
+                Ok(value)
+            }
+            Value::Array(mut array)
+                if array.len() == 3
+                    && matches!(array[2], Value::BulkString(_) | Value::SimpleString(_)) =>
+            {
+                array[2] =
```

Comment:
Does this only convert the third argument to double? Can there be more arguments with more keys where the timeout is greater than the third argument?

Reply:
```
redis> ZADD zset1 0 a 1 b 2 c
(integer) 3
redis> BZPOPMIN zset1 zset2 0
1) "zset1"
2) "a"
3) "0"
```
replay can be one of the following:

[Nil reply](https://redis.io/docs/latest/develop/reference/protocol-spec#bulk-strings): when no element could be popped and the timeout expired.
[Array reply](https://redis.io/docs/latest/develop/reference/protocol-spec#arrays): the keyname, popped member, and its score.

so when its an array replay, which means we popped an element, we want to convert just the score from string to double

Reply:
Gotcha, thanks!

Reply:
np